src/core/ApolloClient.ts

```diff
@@ -468,6 +469,28 @@ export class ApolloClient<TCacheShape> implements DataProxy {
     return this.cache.readQuery<T, TVariables>(options, optimistic);
   }
 
+  /**
+   * Watches the cache store of the fragment according to the options specified
+   * and returns an {@link ObservableQuery}. We can subscribe to this
+   * {@link ObservableQuery} and receive updated results through a GraphQL
+   * observer when the cache store changes.
+   *
+   * You must pass in a GraphQL document with a single fragment or a document
+   * with multiple fragments that represent what you are reading. If you pass
+   * in a document with multiple fragments then you must also specify a
+   * `fragmentName`.
+   *
+   * @param options - An object of type {@link WatchFragmentOptions} that allows
+   * the cache to identify the fragment and optionally specify whether to react
+   * to optimistic updates.
+   */
+
+  public watchFragment<T = any, TVariables = OperationVariables>(
```

Comment:
```suggestion
  public watchFragment<T = unknown, TVariables = OperationVariables>(
```

We should also default to `unknown` here for better type safety.

```suggestion
  public watchFragment<TFragmentData = any, TVariables = OperationVariables>(
```

[nit] What do you think about calling this `TFragmentData` to be a bit more explicit?

Reply:
Updated in https://github.com/apollographql/apollo-client/pull/11465/commits/67c1e1cb18b242644a36b832607b2abaea4a6714